R/req-headers.R

```diff
@@ -47,9 +49,31 @@
 #'    req_headers(!!!headers, HeaderThree = "three") %>%
 #'    req_dry_run()
 #'
-req_headers <- function(.req, ...) {
+#' # Use `.redact` to hide a header in the output
+#' req %>%
+#'   req_headers(Secret = "this-is-private", Public = "but-this-is-not", .redact = "Secret") %>%
+#'   req_dry_run()
+req_headers <- function(.req, ..., .redact = NULL) {
   check_request(.req)
 
-  .req$headers <- modify_list(.req$headers, ...)
+  headers <- list2(...)
+  header_names <- names2(headers)
+  check_redact(.redact)
+
+  redact_out <- attr(.req$headers, "redact") %||% .redact %||% character()
+  redact_out <- union(redact_out, .redact)
+  .req$headers <- modify_list(.req$headers, !!!headers)
+
+  attr(.req$headers, "redact") <- redact_out
+
   .req
 }
+
+check_redact <- function(x, arg = caller_arg(x), call = caller_env()) {
```

Comment:
It might make sense to add the rlang standalone file for type checks in a separate PR.